README.md

```diff
@@ -187,3 +183,126 @@ After that you can start QEMU
+* skl
+    - Recipe: recipes-support/skl/skl_git.bb
+    - Content: Secure Kernel Loader
+* linux-tb
+    - Recipe: recipes-kernel/linux/linux-tb_6.6.bb
+    - Content: Linux kernel
+* grub
+    - Recipe: recipes-bsp/grub/grub_%.bbappend
+* grub-efi
+    - Recipe: recipes-bsp/grub/grub-efi_%.bbappend
+* grub & grub-efi
+    - Recipe: recipes-bsp/grub/grub-tb-common.inc
+    - Content: Common config for both recipes
+
+### Source revision
+
+To change branch or commit used by a recipe you have to change `BRANCH` or
+`SRCREV` variable in appropriate recipe file. In case of Linux kernel those
+variables are named `KBRANCH` and `SRCREV_machine`
+
+### Building modified source
+
+In order to make and test changes to source code you first need to enter kas
+shell
+
+```shell
+kas-container shell meta-trenchboot/kas-generic-tb.yml
+```
+
+After that use `devtool modify <recipe>` to fetch source code of the recipe you
+want to work on. Bitbake will checkout branch and commit set by `BRANCH` and `SRCREV`
+variables.
+
+```shell
+devtool modify skl
+(...)
+INFO: Source tree extracted to /build/workspace/sources/skl
+INFO: Using source tree as build directory since that would be the default for this recipe
+INFO: Recipe skl now set up to build from /build/workspace/sources/skl
+```
+
+All recipes' sources you wish to modify will be in `/build/workspace/sources`.
+After modifications, you can try to a build recipe by using
+`devtool build <recipe>` or if you want to build the whole image with your changes
+then use `devtool build-image tb-minimal-image`.
+After building the image, you can [install](#flash) and [boot](#booting) it or
+run it in [QEMU](#running-in-qemu).
+
+#### GRUB
+
+Currently, if you try to build `grub` or `grub-efi` twice you will get error
+
+```text
+| Updating file build-aux/config.rpath (backup in build-aux/config.rpath~)
+| Updating file m4/extern-inline.m4 (backup in m4/extern-inline.m4~)
+| gnulib/gnulib-tool: *** file /build/workspace/sources/grub/gnulib/m4/lib-ld.m4 not found
+| gnulib/gnulib-tool: *** Stop.
+| /build/workspace/sources/grub/bootstrap: gnulib-tool failed
+| WARNING: exit code 1 from a shell command.
+```
+
+To fix this you have to delete `/build/workspace/sources/<recipe>/gnulib`
+directory before building recipe again.
+
+### Local files
+
+Files added by Yocto recipe are stored inside `sources/<recipe>/oe-local-files`
+folder. Example of local file is `defconfig` file in `linux-tb` recipe
+
+### Deployment
+
+To deploy component to target machine after making changes you can use rsync
+with path depending on component you modified. `<destination>` can be either
+`<USER>@<IP>:/` or path to mounted rootfs partition. Similarly,
+`<boot_destination>` is `<USER>@<IP>:/boot` or path to mounted boot partition.
+To deploy files via ssh you need to start/enable ssh server on target machine.
+
+* grub
+
+    ```shell
+    rsync -chavP --stats /build/tmp/work/core2-64-tb-linux/grub/2.06/image/ <destination>
```

Comment:
How about moving these command into some script?

I understand that the `devtool deploy` does not work for us in most cases?

Reply:
Not really except maybe only for `grub-efi` (if we ignore `grub.cfg`).
I added script here 4550ec7d4798781f66eb42819599e6b5b6fc9780 but it would need to be tested more thoroughly especially deployment (if all needed files are copied).
There is still `grub` deployment to do.